### ISSUE TITLE
Pipe colors no longer appear before some fluid put in system

### ISSUE THREAD

**Issue by ghost** [raiguard/PipeVisualizer#23]
In previous versions, the visualizer was able to detect the fluid outputs/inputs of the connected machines, it looks like this functionality was removed since the highlighted boxes for containers was added. It makes planning out the pipe layout for the initial blueprint/stamp more difficult as it now requires some input to be fed into the system before it gets colored in, which may get wasted in the designing process. 

Could you add the machine's i/o detection back?

Lack of color while designing:
![nocolor](https://github.com/raiguard/PipeVisualizer/assets/75500666/539a54d9-7ad4-49cd-b45c-0a591f065833)

Has color while fed, and what it used to do in previous version before any input too:
![withcolor](https://github.com/raiguard/PipeVisualizer/assets/75500666/d18fb33d-ca87-4244-86e4-c5cd09e6ea1c)

This mod has been a godsend, but it would have been much harder to design my pipe spaghetti if there was no color while I was making it, as I would need to keep going back to the source to find what type of fluid pipes are carrying. Pic of spaghetti:
![largecolor](https://github.com/raiguard/PipeVisualizer/assets/75500666/6832e4f7-d6b0-4e6f-9058-a52a30d8bea9)

**Comment by dlathro1** [comment]
I would like to second this post. I find the newest version of this mod borderline unusable when designing blueprints.

**Comment by Zmiy69** [comment]
I also find it VERY inconvenient that pipes are not immediately painted in the color of the liquid, as it was in the previous version.

**Comment by martin-t** [comment]
For me this is also a massive downgrade. I'll give 2.1 a bit of time but mostly only to find out if the previous versions were the reason for the game randomly getting stuck. If not, then this'll make me downgrade to 1.0. It's a really important feature.

**Comment by raiguard** [comment]
Fixing this properly will require a new API addition. I could work around it but it would make for a ton of ugly spaghetti code.

**Comment by whizzball1** [comment]
Does 'color by system' solve this problem (with the downside of the colors not representing the fluids)?

**Comment by Nagapito** [comment]
Yeah, like @whizzball1 said, for this situation, using ctrl+h to switch from color by fluid to color by system seems to be an acceptable solution. 
Since pipe network system is diferente, the colors will change unless you merge the networks, which is also a good giveway that a mistake was done!
